camel/models/open_source_model.py

```diff
@@ -0,0 +1,140 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from types import GeneratorType
+from typing import Any, Dict, List
+
+from camel.configs import OPENAI_API_PARAMS
+from camel.messages import OpenAIMessage
+from camel.models import BaseModelBackend
+from camel.typing import ModelType
+from camel.utils import BaseTokenCounter, OpenSourceTokenCounter
+
+
+class OpenSourceModel(BaseModelBackend):
+    r"""Class for interace with OpenAI-API-compatible servers running
+    open-source models.
+    """
+
+    def __init__(
+        self,
+        model_type: ModelType,
+        model_config_dict: Dict[str, Any],
+    ) -> None:
+        r"""Constructor for model backends of Open-source models.
+
+        Args:
+            model_type (ModelType): Model for which a backend is created.
+            model_config_dict (Dict[str, Any]): A dictionary that will
+                be fed into :obj:`openai.ChatCompletion.create()`.
+        """
+        super().__init__(model_type, model_config_dict)
+
+        # Check whether the input model type is open-source
+        if not model_type.is_open_source:
+            raise ValueError(
+                f"Model `{model_type}` is not a supported open-source model")
+
+        # Check whether input model path is empty
+        model_path: str | None = self.model_config_dict.get("model_path", None)
```

Comment:
We do not support this style of type hints. Please rework for python 3.8 compatibility.